language/module.md

```diff
@@ -0,0 +1,71 @@
+* `/Dockerfile`
+
+Note that above are just examples and not a complete list. The goal of this
+document is to provide an allowlist, not a denylist.
+
+The official
+[Puppet documentation](https://www.puppet.com/docs/puppet/latest/modules_fundamentals.html)
+already explains what a module is and what it can contain.
+
+| Directories and Files | Purpose |
+|-----------------------|---------|
+| `/manifests/`         | contains Puppet code |
+| `hiera.yaml`          | Can define a Hiera configuration for Hiera data within this module |
+| `/data/`              | If the module has a `hiera.yaml`, the related data has to be within `/data/` |
+| `/templates/`         | Stores [epp](https://www.puppet.com/docs/puppet/latest/lang_template_epp.html) or [erb](https://www.puppet.com/docs/puppet/latest/lang_template_erb.html) templates |
+| `/files/`             | Static files that Puppet code within the module will distribute |
+| `/examples/`          | Example Puppet snippets that explain how to use the module. They can be used within acceptance tests |
+| `/facts.d/`           | [External facts](https://www.puppet.com/docs/puppet/latest/external_facts.html) that are synced via [pluginsync](https://www.puppet.com/docs/puppet/latest/plugins_in_modules.html) |
+| `/lib/facter/`        | Contains custom facts |
+| `/lib/puppet/type/`   | Custom types |
+| `/lib/puppet/provider/` | Custom provider for one or multiple types |
+| `/lib/puppet/functions/` | Modern functions in Ruby for the new API |
+| `/lib/puppet/parser/functions/` | Legacy functions in Ruby |
+| `/lib/puppet_x/`      | Custom Ruby modules to extend types, providers, functions or facts |
+| `/lib/augeas/lenses/` | Custom [Augeas](https://augeas.net/) lenses |
+| `/functions/`         | Can contain functions written in Puppet DSL |
+| `/metadata.json`      | Metadata file that describes the module based on [a schema](https://www.puppet.com/docs/puppet/latest/modules_metadata.html) |
+| `/README`             | A README that describes what the module does. It's best practice to add a suffix like `.txt`, `.md`, `.rst` for the used markup language |
+| `/LICENSE`            | `/metadata.json` specifies a license. The license text needs to be distributed in a LICENSE file. It can have a suffix if a markup language is used |
+| `/CHANGELOG`          | A Changelog that's updated for every release. A new release should not alter existing changelog entries. It can have a suffix if a markup language is used |
+| `/HISTORY`            | The history file can contain old changelogs. Useful when the tool used to generate changelogs is changed or a module is migrated to a new namespace |
+| `/docs/`              | Directory for additional documentation |
+| `/REFERENCE.md`       | [puppet-strings](https://www.puppet.com/docs/puppet/latest/puppet_strings.html) based documentation in markdown, updated on each release |
+| `/locales/`           | Used for i18n support, can contain translated strings |
```

Comment:
Isn't this deprecated by now? https://github.com/puppetlabs-toy-chest/puppetlabs-translate is archived in the toy chest.

Reply:
I thought so as well last week, but I'm unsure. @joshcooper do you have an idea?

Reply:
This was archived a long time ago.
As far as I know there are no plans to revisit it as I remember removing the compatible code for it.

Reply:
@joosh in https://github.com/puppetlabs/puppet-specifications/pull/157#discussion_r1373898104 mentioned the settings and at least those are still present in puppet and puppetserver. Maybe it's time to clean that up.

Reply:
Interesting, guess it was left in place....

Reply:
In my experience it's much easier to add a feature than to remove it. Finding what's missing is easy. Finding what's used only by a certain feature is much harder.

Reply:
We should keep it for now because it's still possible to use it. I added a note that the module schema will be updated to a new version and `/locales/` removed when it's actually dropped from puppet and puppetserver.